comp/core/workloadmeta/collectors/util/image_util.go

```diff
@@ -32,18 +37,52 @@ func UpdateSBOMRepoMetadata(sbom *workloadmeta.SBOM, repoTags, repoDigests []str
 	}
 
 	properties := *sbom.CycloneDXBOM.Metadata.Component.Properties
-	properties = removeOldRepoProperties(properties)
+	mismatched := !propertiesEqualsValues(properties, repoTags, repoTagPropertyKey) || !propertiesEqualsValues(properties, repoDigests, repoDigestPropertyKey)
+
+	if mismatched {
+		sbom = deepcopy.Copy(sbom).(*workloadmeta.SBOM)
+		newProperties := *sbom.CycloneDXBOM.Metadata.Component.Properties
+		newProperties = cleanProperties(newProperties)
 
-	properties = appendRepoProperties(properties, repoTags, repoTagPropertyKey)
-	properties = appendRepoProperties(properties, repoDigests, repoDigestPropertyKey)
+		newProperties = appendProperties(newProperties, repoTags, repoTagPropertyKey)
+		newProperties = appendProperties(newProperties, repoDigests, repoDigestPropertyKey)
 
-	sbom.CycloneDXBOM.Metadata.Component.Properties = &properties
+		// Sort properties to ensure consistent ordering for tests
+		sort.Slice(newProperties, func(i, j int) bool {
+			return newProperties[i].Name < newProperties[j].Name || newProperties[i].Value < newProperties[j].Value
+		})
```

Comment:
In order to be able to properly sort a slice, the algorithm must be given a partial order.
The function given here isn’t a partial order.

Here is a counter-example of violation of the antisymmetry:
```go
newProperties = [
  {
    Name: "A",
    Value: "B",
  },
  {
    Name: "B",
    Value: "A",
  },
}
```

`f(0, 1) == true` because `newProperties[0].Name /* A */ < newProperties[1].Name /* B */`
`f(1, 0) == true` as well because `newProperties[1].Value /* A */ < newProperties[0].Value /* B */`
although `newProperties[0] ≠ newProperties[1]`

Reply:
Nice catch, I fixed it by moving down the sort func